recurring_contract/model/contract_group.py

```diff
@@ -82,19 +41,59 @@ def _set_last_paid_invoice(self):
     payment_term_id = fields.Many2one('account.payment.term',
                                       _('Payment Term'),
                                       track_visibility="onchange")
-
     next_invoice_date = fields.Date(
         compute='_set_next_invoice_date',
         string=_('Next invoice date'), store=True)
-
     last_paid_invoice_date = fields.Date(
         compute='_set_last_paid_invoice',
         string=_('Last paid invoice date'))
 
     change_method = fields.Selection(
-        selection=__get_change_methods, default='do_nothing',
-        string=_('Change method'))
+        selection=lambda self: self.__get_change_methods(),

```

Comment:
fields selection can have the method given in quotes so you can avoid lambda. keyword selection is not mandatory, as the first parameter is the function.